python/metrics/src/hydrotools/metrics/flow_duration_curve.py

```diff
@@ -0,0 +1,390 @@
+    chosen.
+        This method uses the Stationary Bootstrap method, which is suitable for
+    time series of streamflow that may have a high degree of time-dependent
+    autocorrelation. This uses a method that automatically determines the
+    optimal block size. This optimal block size will tend toward 1 for
+    uncorrelated data and is suitable for series of independent values as well
+    (e.g. annual peaks).
+
+    Parameters
+    ----------
+    series: array-like, required
+        An array of numerical streamflow values used to generate a flow
+        duration curve.
+    quantiles: array-like, required
+        Passed directly to `numpy.quantiles`. Specifies the quantiles to sample
+        from the bootstrapped posterior distribution. For a two-tailed 95%
+        confidence interval use `quantiles=[0.025, 0.975]`.
+    repetitions: int, optional, default 1200
+        Number of bootstrapped samples in the posterior distribution of values
+        used to extract quantiles.
+    fdc_generator: FDCGenerator, optional
+        A callable that takes a series and generates a flow duration curve.
+        Defaults to `empirical_flow_duration_curve`.
+    fdc_params: optional
+        Additional keyword arguments passed directly to `fdc_generator`.
+    
+    Returns
+    -------
+    probabilities: numpy.ndarray
+        Array of values that represent exceedance probabilities, also called
+        exceedance frequencies. The range of possible values is [0.0, 1.0].
+        These values usually represent the "x-values" on a flow duration curve
+        plot. These values can be converted into recurrence intervals by
+        dividing them into 1.0 (i.e. 1.0 / probabilities). Note that these
+        recurrence intervals will reflect the sampling frequency of the input
+        series (e.g. "hourly", "daily", or "annual" recurrence intervals).
+    values: numpy.ndarray
+        Array of actual streamflow values in the same units as the input
+        series. These values correspond directly to exceedance frequencies in
+        probabilities. These values usually represent the "y-values" on a flow
+        duration curve plot. values will have a shape (m, n) where m is
+        the number of `quantiles` specified and n is the number of values in 
+        probabilities.
+    """
+    # Validate
+    size = len(series)
+    assert size > 0, "series is empty"
+    assert repetitions > 0, "repetitions must be greater than 0"
+
+    # Setup bootstrap
+    block_size = optimal_block_length(series)["stationary"].iloc[0]
+    bs = StationaryBootstrap(block_size, series, seed=2025)
+
+    # Bootstrap values
+    def value_generator(s):
+        return fdc_generator(s, **fdc_params)[1]
+    posterior = bs.apply(value_generator, repetitions)
+
+    # Extract quantiles
+    return (
+        fdc_generator(series, **fdc_params)[0],
+        np.quantile(posterior, quantiles, axis=0)
+        )
+
+def exceedance_values(
```

Comment:
Even though the method only contains a one line method of numpy, I do like that it contextualizes the arguments and what it's trying to accomplish.  I would imagine most of your users would easily be able to find out what this method is doing by reading your docs and the docstring, however since np.interp is the only method called perhaps include the word interpolate or interp in the method would make it immediately clear.